dlt/common/destination/reference.py

```diff
@@ -498,6 +503,30 @@ def should_truncate_table_before_load_on_staging_destination(self, table: TTable
         return True
 
 
+class SupportsDataAccess(ABC):
+    """Add support for accessing data as arrow tables or pandas dataframes"""
+
+    @abstractmethod
+    def iter_df(
+        self,
+        *,
+        table: str = None,
+        batch_size: int = 1000,
+        sql: str = None,
+        prepare_tables: List[str] = None,
+    ) -> Generator[DataFrame, None, None]: ...
+
+    @abstractmethod
+    def iter_arrow(
```

Comment:
It's probably a good idea to support `pyarrow.RecordBatchReader` alongside `pyarrow.Table` for larger-than-memory data.

Or just expose a `pyarrow.Dataset`, from which the user can create either a `pyarrow.RecordBatchReader` or `pyarrow.Table`.

Reply:
Note: using Dataset possible allows us to "mount" any external table as a cursor that reads data so we do not need to download full table locally. we may be able to ie. mount snowflake or bigquery tables are duckdb tables